README.md

```diff
@@ -215,3 +192,379 @@ if available(.settingsFeature) {
+final class BuyMockService: BuyService {}
+```
+
+
+If it is known in advance that only classes will conform to the protocol, you need to add the keyword AnyObject:
+
+🟢
+```
+protocol SomeProtocol: AnyObject {}
+```
+
+🔴
+```
+protocol SomeProtocol {}
+```
+
+
+### Class structure
+We try to avoid inheritance. For all classes from which inheritance is not planned, the final keyword must be explicitly specified. This speeds up the dispatching of calls in runtime and makes life easier for the compiler.:
+
+```
+final class BuyView: UIView {}
+```
+
+
+Dependencies and properties are always written at the top of the class, indicated by a comment without indentation after:
+
+🟢
+```
+final class SomeClass {
+
+    // Dependencies
+    private let buyService: BuyService
+    private let sellService: SellService
+
+    // Private properties
+    private var amount: Double?
+}
+```
+
+🔴
+```
+final class SomeClass {
+
+    // Dependencies
+
+    private let buyService: BuyService
+    private let sellService: SellService
+
+    // Private properties
+
+    private var amount: Double?
+}
+```
+
+
+All other elements are indicated with // MARK: - indented after:
+
+🟢
+```
+final class SomeClass {
+
+    // MARK: - Init
+
+    public init() {...}
+
+    // MARK: - BuyService
+
+    func buyCrypto(_ crypto: Crypto) {...}
+}
+```
+
+🔴
+```
+final class SomeClass {
+
+    // MARK: - Init
+    public init() {...}
+
+    // MARK: - BuyService
+    func buyCrypto(_ crypto: Crypto) {...}
+}
+```
+
+
+If a large number of private functions are typed, then you should not dump them all in one pile, you need to break them into logically connected blocks.
+
+🟢 Division within the class:
+```
+final class SomeClass {
+
+    // MARK: - Listeners
+
+    private func addListeners() {...}
+    private func removeListeners() {...}
+    private func removeListener(_ listener: Listener) {...}
+
+    // MARK: - Actions
+
+    @objc private func didTapClose() {...}
+}
+```
+
+
+🟢 Takeaway in extensions:
+```
+final class SomeClass {
+    ...
+}
+
+// MARK: - Listeners
+
+extension SomeClass {
+    private func addListeners() {...}
+    private func removeListeners() {...}
+    private func removeListener(_ listener: Listener) {...}
+}
+
+// MARK: - Actions
+
+extension SomeClass {
+   @objc private func didTapClose() {...}
+}
+```
+
+🔴
+```
+final class SomeClass {
+
+    // MARK: - Private
+
+    func addListeners() {...}
+    @objc func didTapClose() {...}
+    func removeListeners() {...}
+    func removeListener(_ listener: Listener) {...}
+}
+```
+
+
+There should be no line break before the closing brackets:
+
+🟢
+```
+struct SomeStruct {
+    func listen() {
+        ...
+    }
+}
+```
+
+🔴
+```
+struct SomeStruct {
+    func listen() {
+        ...
+    }
+
+}
+```
+
+
+### Switch statement
+For enum we don't use default in switch. When changing the enum during assembly, all the places where it is used will be immediately visible:
```

Comment:
I think we should TRY TO AVOID using default, but we CAN NOT completely omit the default. In some cases, default might be unavoidable.

Reply:
In my opinion it's better to stay it here and have it as optional